clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -163,26 +213,40 @@ public void authenticationFailed(String id, long now, AuthenticationException ex
     }
 
     /**
-     * Return true if the connection is ready.
+     * Return true if the connection is in the READY state and currently not throttled.
+     *
      * @param id the connection identifier
+     * @param now the current time
      */
-    public boolean isReady(String id) {
+    public boolean isReady(String id, long now) {
         NodeConnectionState state = nodeState.get(id);
-        return state != null && state.state == ConnectionState.READY;
+        return state != null && state.state == ConnectionState.READY && !isThrottled(id, now);
     }
 
     /**
-     * Return true if there is at least one node with connection in ready state and false otherwise.
+     * Return true if there is at least one node with connection in the READY state and not throttled. Returns false
+     * otherwise.
+     *
+     * @param now the current time
      */
-    public boolean hasReadyNodes() {
+    public boolean hasReadyNodes(long now) {
         for (Map.Entry<String, NodeConnectionState> entry : nodeState.entrySet()) {
             NodeConnectionState state = entry.getValue();
-            if (state != null && state.state == ConnectionState.READY)
+            if (state != null && state.state == ConnectionState.READY && state.throttleDeadlineMs <= now)
                 return true;
         }
         return false;
     }
 
+    /**
+     * Return true if the connection has been established
+     * @param id The id of the node to check
+     */
+    public boolean isConnected(String id) {
```

Comment:
Now that isReady() checks for throttling status, I don't think it can be used directly for pollDelayMs(). pollDelayMs() needs to check if the state is READY and *throttled*. I can directly check if the state is READY, instead of isConnected(). Will that work?